scss/_patterns_chip.scss

```diff
@@ -1,90 +1,170 @@
 @import 'settings';
 
 @mixin vf-p-chip {
-  $chip-line-height: 1rem;
-  $base-background-opacity-amount: 0.05;
-  $color-background-base: adjust-color($color-x-dark, $lightness: 100% * (1 - $base-background-opacity-amount));
-  $color-background-hover: adjust-color($color-x-dark, $lightness: 100% * (1 - ($base-background-opacity-amount + $hover-background-opacity-amount)));
+  @include vf-chip-default;
+  @include vf-chip-positive;
+  @include vf-chip-caution;
+  @include vf-chip-negative;
+  @include vf-chip-information;
+}
 
+@mixin vf-chip-default {
   .p-chip {
-    @extend %small-text;
+    @include vf-chip-base();
+  }
+}
+
+@mixin vf-chip-positive {
+  .p-chip--positive {
+    @include vf-chip-base(positive);
+  }
+}
+
+@mixin vf-chip-caution {
+  .p-chip--caution {
+    @include vf-chip-base(caution);
+  }
+}
+
+@mixin vf-chip-negative {
+  .p-chip--negative {
+    @include vf-chip-base(negative);
+  }
+}
+
+@mixin vf-chip-information {
+  .p-chip--information {
+    @include vf-chip-base(information);
+  }
+}
+
+@mixin vf-chip-base($type: neutral) {
+  $chip-line-height: 1rem;
+  $color-background: null;
+  $color-border: null;
+
+  @if ($type == negative) {
+    $color-background: map-get($colors-light-theme--tinted-backgrounds, negative);
```

Comment:
If "negative" is in `$type` variable, wouldn't simply passing it to map work?

```scss
$color-background: map-get($colors-light-theme--tinted-backgrounds, $type);
$color-border: map-get($colors-light-theme--tinted-borders, $type);
```

Reply:
yes, didn't think of that. thanks.